as-vending/routes/controller.go

```diff
@@ -0,0 +1,230 @@
+// Copyright © 2022 Intel Corporation. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+package routes
+
+import (
+	"as-vending/functions"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	utilities "github.com/intel-iot-devkit/automated-checkout-utilities"
+)
+
+type Controller struct {
+	lc           logger.LoggingClient
+	service      interfaces.ApplicationService
+	vendingState functions.VendingState
+}
+
+func NewController(lc logger.LoggingClient, service interfaces.ApplicationService, vendingState functions.VendingState) Controller {
+	return Controller{
+		lc:           lc,
+		service:      service,
+		vendingState: vendingState,
+	}
+}
+func (c *Controller) GetVendingState() *functions.VendingState {
+	return &c.vendingState
+}
+
+func (c *Controller) AddAllRoutes() error {
+	var err error
+
+	err = c.service.AddRoute("/boardStatus", c.BoardStatus, http.MethodPost)
+	if errWithMsg := c.errorAddRouteHandler(err); errWithMsg != nil {
+		return errWithMsg
+	}
+
+	err = c.service.AddRoute("/resetDoorLock", c.ResetDoorLock, http.MethodPost)
+	if errWithMsg := c.errorAddRouteHandler(err); errWithMsg != nil {
+		return errWithMsg
+	}
+
+	err = c.service.AddRoute("/maintenanceMode", c.GetMaintenanceMode, http.MethodGet)
+	if errWithMsg := c.errorAddRouteHandler(err); errWithMsg != nil {
+		return errWithMsg
+	}
+
+	return nil
+
+}
+
+// GetMaintenanceMode will return a JSON response containing the boolean state
+// of the vendingState's maintenance mode.
+func (c *Controller) GetMaintenanceMode(writer http.ResponseWriter, req *http.Request) {
+	utilities.ProcessCORS(writer, req, func(writer http.ResponseWriter, req *http.Request) {
+		mm, _ := utilities.GetAsJSON(functions.MaintenanceMode{MaintenanceMode: c.vendingState.MaintenanceMode})
+		utilities.WriteJSONHTTPResponse(writer, req, http.StatusOK, mm, false)
+	})
+}
+
+func (c *Controller) errorAddRouteHandler(err error) error {
+	if err != nil {
+		c.lc.Errorf("error adding route: %s", err.Error())
+		return fmt.Errorf("error adding route: %s", err.Error())
```

Comment:
could use error msg variable to reduce the duplicate hard code err message

Reply:
updated